lib/Parse/ParseDecl.cpp

```diff
@@ -2473,8 +2473,9 @@ Parser::parseMacroRoleAttribute(
 /// \returns \c None if an error was diagnosed; \c Identifier() if the argument list was permissibly
 ///          omitted; the identifier written by the user otherwise.
 static std::optional<Identifier> parseSingleAttrOptionImpl(
-    Parser &P, SourceLoc Loc, SourceRange &AttrRange, StringRef AttrName,
-    DeclAttrKind DK, bool allowOmitted, DiagRef nonIdentifierDiagnostic) {
+    Parser &P, SourceLoc Loc, SourceRange &AttrRange, StringRef AttrName, DeclAttrKind DK,
+    ParserStatus &Status, bool allowOmitted, DiagRef nonIdentifierDiagnostic,
+    llvm::function_ref<void()> codeCompletionCallback = {}) {
```

Comment:
Sorry, I could have noticed this in the last review round but I think this could be simplified if, instead of the `codeCompletionCallback`, we would pass in a `std::optional<ParameterizedDeclAttributeKind>` and call the `CodeCompletionCallbacks` from in here. What do you think?

Reply:
Yes, I think this simplifies the code a bit more. Fixed. 👍🏼